wenet/dataset/processor.py

```diff
@@ -292,24 +292,32 @@ def tokenize(data, symbol_table, bpe_model=None, split_with_space=False):
         Returns:
             Iterable[{key, wav, txt, tokens, label, sample_rate}]
     """
-    # TODO(Binbin Zhang): Support BPE
     if bpe_model is not None:
         import sentencepiece as spm
         sp = spm.SentencePieceProcessor()
         sp.load(bpe_model)
     for sample in data:
         assert 'txt' in sample
-        txt = sample['txt']
+        txt = sample['txt'].strip()
         label = []
         tokens = []
         if bpe_model is not None:
-            txt = bpe_preprocess(txt)
-            mix_chars = seg_char(txt)
+            # CJK(China Japan Korea) unicode range is [U+4E00, U+9FFF], ref:
+            # https://en.wikipedia.org/wiki/CJK_Unified_Ideographs_(Unicode_block)
+            pattern = re.compile(r'([\u4e00-\u9fff])')
+            # Example:
+            #   txt   = "你好 ITS'S OKAY 的"
+            #   chars = ["你", "好", " ITS'S OKAY ", "的"]
+            chars = pattern.split(txt.upper())
+            mix_chars = [w for w in chars if len(w.strip()) > 0]
             for j in mix_chars:
```

Comment:
j is for index in the most cases.  ch or w is better.

Reply:
done